noxfile.py

```diff
@@ -92,6 +93,9 @@ def tests(session, shard):
         env["MISTRAL_API_KEY"] = session.env.get("MISTRAL_API_KEY")
         env["OPENAI_API_KEY"] = session.env.get("OPENAI_API_KEY")
 
+    if shard == "openai_agents":
+        env["OPENAI_API_KEY"] = session.env.get("OPENAI_API_KEY", "DUMMY_API_KEY")
```

Comment:
This never reads from the env variable even when explicitly passed which breaks re-recording